src/chat/model/CTECTwitter.java

```diff
@@ -0,0 +1,256 @@
+package chat.model;
+
+import chat.controller.ChatController;
+import twitter4j.TwitterException;
+import twitter4j.TwitterFactory;
+import twitter4j.Twitter;
+import twitter4j.Status;
+import java.util.List;
+import java.util.Scanner;
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import twitter4j.Paging;
+
+public class CTECTwitter 
+{
+	private ChatController baseController;
+	private Twitter chatbotTwitter;
+	private List<Status> searchedTweets;
+	private List<String> tweetedWords;
+	private List<List<String>> rankedWords;
+	
+	public CTECTwitter(ChatController baseController)
+	{
+		this.baseController = baseController;
+		searchedTweets = new ArrayList<Status>();
+		tweetedWords = new ArrayList<String>();
+		rankedWords = new ArrayList<List<String>>();
+		this.chatbotTwitter = TwitterFactory.getSingleton();
+		
+	}
+	
+	public void sendTweet(String textToTweet)
+	{
+		try
+		{
+			chatbotTwitter.updateStatus(textToTweet + "#JavaChatbot @ChatbotCTEC");
+		}
+		catch(TwitterException tweetError)
+		{
+			baseController.handleErrors(tweetError);
+		}
+		catch(Exception otherError)
+		{
+			baseController.handleErrors(otherError);
+		}
+	}
+	
+	private String [] createIgnoredWordArray()
+	{
+		String [] boringWords;
+		
+		int wordCount = 0;
+		Scanner wordScanner = new Scanner(this.getClass().getResourceAsStream("commonWords.txt"));
+		
+		while(wordScanner.hasNextLine())
+		{
+			wordScanner.nextLine();
+			wordCount++;
+		}
+		
+		boringWords = new String [wordCount];
+		wordScanner.close();
+		
+		wordScanner = new Scanner(this.getClass().getResourceAsStream("commonWords.txt"));
+		for(int index = 0; index < boringWords.length; index ++)
+		{
+			boringWords[index] = wordScanner.nextLine();
+		}
+		
+		wordScanner.close();
+		return boringWords;
+	}
+	
+	private void collectTweets(String username)
+	{
+		searchedTweets.clear();
+		tweetedWords.clear();
+		
+		Paging statusPage = new Paging(1, 100);
+		int page = 1;
+		
+		while(page <= 10)
+		{
+			statusPage.setPage(page);
+			try
+			{
+				searchedTweets.addAll(chatbotTwitter.getUserTimeline(username, statusPage));
+				
+				page ++;
```

Comment:
It looks like that the variable `page` will be increased in both branches if the `try-catch` flow, so that it could be brought outside of `try-catch`